test/gtest/activ_common.hpp

```diff
@@ -0,0 +1,532 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#pragma once
+
+#include <cmath>
+#include <future>
+#include <numeric>
+#include <stdexcept>
+#include <thread>
+
+#include <miopen/miopen.h>
+
+#include "../tensor_holder.hpp"
+
+namespace miopen {
+namespace tests {
+
+namespace activ_func {
+
+class ActivationPASTHRU
+{
+public:
+    template <class T, class Tparam>
+    static T Forward(Tparam alpha, Tparam beta, Tparam gamma, T x)
+    {
+        return x;
+    }
+
+    template <class T, class Tparam>
+    static T Backward(Tparam alpha, Tparam beta, Tparam gamma, T dy, T x, T y)
+    {
+        return dy;
+    }
+};
+
+class ActivationLOGISTIC
+{
+public:
+    template <class T, class Tparam>
+    static Tparam Forward(Tparam alpha, Tparam beta, Tparam gamma, T x)
+    {
+        return 1 / (1 + std::exp(-x));
+    }
+
+    template <class T, class Tparam>
+    static Tparam Backward(Tparam alpha, Tparam beta, Tparam gamma, T dy, T x, T y)
+    {
+        return static_cast<Tparam>(dy) * y * (static_cast<Tparam>(1) - y);
+    }
+};
+
+class ActivationTANH
+{
+public:
+    template <class T, class Tparam>
+    static Tparam Forward(Tparam alpha, Tparam beta, Tparam gamma, T x)
+    {
+        return beta * std::tanh(alpha * x);
+    }
+
+    template <class T, class Tparam>
+    static Tparam Backward(Tparam alpha, Tparam beta, Tparam gamma, T dy, T x, T y)
+    {
+        return dy * alpha * (beta - static_cast<Tparam>(y) * y / beta);
+    }
+};
+
+class ActivationRELU
+{
+public:
+    template <class T, class Tparam>
+    static T Forward(Tparam alpha, Tparam beta, Tparam gamma, T x)
+    {
+        return (x > static_cast<Tparam>(0)) ? x : static_cast<T>(0);
```

Comment:
```suggestion
        return (x > static_cast<T>(0)) ? x : static_cast<T>(0);
```
At least `x` is a `T`, not `Tparam` and casting `x` to `Tparam` does not make much sense for the comparison to 0.